cmd/kube-controller-manager/app/plugins_providers.go

```diff
@@ -35,40 +35,47 @@ import (
 
 type probeFn func() []volume.VolumePlugin
 
-func appendPluginBasedOnMigrationFeatureFlags(plugins []volume.VolumePlugin, inTreePluginName string, featureGate featuregate.FeatureGate, pluginMigration, pluginMigrationComplete featuregate.Feature, fn probeFn) ([]volume.VolumePlugin, error) {
+func appendPluginBasedOnFeatureFlags(plugins []volume.VolumePlugin, inTreePluginName string, featureGate featuregate.FeatureGate, pluginInfo pluginInfo) ([]volume.VolumePlugin, error) {
 	// Skip appending the in-tree plugin to the list of plugins to be probed/initialized
 	// if the CSIMigration feature flag and plugin specific feature flag indicating
 	// CSI migration is complete
-	err := csimigration.CheckMigrationFeatureFlags(featureGate, pluginMigration, pluginMigrationComplete)
+	migrationComplete, err := csimigration.CheckMigrationFeatureFlags(featureGate, pluginInfo.pluginMigrationFeature,
+		pluginInfo.pluginMigrationCompleteFeature, pluginInfo.pluginUnregisterFeature)
 	if err != nil {
 		klog.Warningf("Unexpected CSI Migration Feature Flags combination detected: %v. CSI Migration may not take effect", err)
 		// TODO: fail and return here once alpha only tests can set the feature flags for a plugin correctly
 	}
-	if featureGate.Enabled(features.CSIMigration) && featureGate.Enabled(pluginMigration) && featureGate.Enabled(pluginMigrationComplete) {
-		klog.Infof("Skip registration of plugin %s since feature flag %v is enabled", inTreePluginName, pluginMigrationComplete)
+	if migrationComplete {
```

Comment:
Do we need to check migration feature gate at all? Should it only be the Unregister feature gate that is used?

Reply:
Yeah... we will remove this after CSIMigrationvSphereComplete flag is removed. Otherwise yes, we do need to check because a combination of CSIMigrationvSphereComplete && CSIMigrationvSphere && CSIMigration will also lead to the unregister of the in-tree plugin

Reply:
Ok can you add a comment that this check will be removed?

Reply:
Done